### PR TITLE
Task 7 forms

### PR DESCRIPTION
**Pull request by LenkouAndrei**

### CODE REVIEW

File: movies-manager/src/containers/movie-form/movie-form.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+    return <Formik
+        initialValues={{ ...movie, genres }}
+        validationSchema={Yup.object({
+            title: Yup.string()
+                .min(3, 'Must be 3 characters or more')
+                .required('Required'),
+            release_date: Yup.string()
+                .matches(
+                    /(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])/,
```
Comment (Vladlan) [inline movies-manager/src/containers/movie-form/movie-form.tsx:74]: Find better regex:
validation_problem_lenkov

Comment (Vladlan) [reply movies-manager/src/containers/movie-form/movie-form.tsx:74]: Also, I know this was not part of requirements, but add smth like https://material-ui.com/ru/components/snackbars/ (you can use any library or write your custom component to display error messages).

**Review by Vladlan — CHANGES_REQUESTED** [review]

File: movies-manager/src/containers/movie-form/movie-form.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                )
+                .required('Required'),
+            poster_path: Yup.string()
+                .matches(
+                    /^(?:(?<scheme>[^:\/?#]+):)?(?:\/\/(?<authority>[^\/?#]*))?(?<path>[^?#]*\/)?(?<file>[^?#]*\.(?<extension>[Jj][Pp][Ee]?[Gg]|[Pp][Nn][Gg]|[Gg][Ii][Ff]))(?:\?(?<query>[^#]*))?(?:#(?<fragment>.*))?$/,
+                    'Must be in a predefine format'
+                )
+                .required('Required'),
+            overview: Yup.string()
+                .min(5, 'Tesxt should be in a range from 5 to 250 symbols')
```
[the remark this answers is not in the document]
Comment (LenkouAndrei) [reply movies-manager/src/containers/movie-form/movie-form.tsx:86]: Done

File: movies-manager/src/containers/movie-form/movie-form.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    return <Formik
+        initialValues={{ ...movie, genres }}
+        validationSchema={Yup.object({
+            title: Yup.string()
+                .min(3, 'Must be 3 characters or more')
+                .required('Required'),
+            release_date: Yup.string()
+                .matches(
+                    /(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])/,
+                    // tslint:disable-next-line
```
[the remark this answers is not in the document]
Comment (LenkouAndrei) [reply movies-manager/src/containers/movie-form/movie-form.tsx:75]: Done use \ before "

File: movies-manager/src/types/types.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ export interface IMovieInfo {
     overview: string;
     budget: number;
     revenue: number;
-    genres: string[];
+    genres: T;
```
[the remark this answers is not in the document]
Comment (LenkouAndrei) [reply movies-manager/src/types/types.ts:13]: Done

File: movies-manager/src/containers/movie-form/movie-form.tsx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+                {movie.id && FORM_HEADLINES.EDIT_MOVIE || FORM_HEADLINES.ADD_MOVIE}
+            </h2>
+            { movieIdField }
+            <InputControl
+                label='Title'
+                name='title'
+            />
+            <InputControl
+                label='release date'
+                name='release_date'
```
[the remark this answers is not in the document]
Comment (LenkouAndrei) [reply movies-manager/src/containers/movie-form/movie-form.tsx:129]: Done

File: movies-manager/src/containers/movie-form/movie-form.tsx (hunk 1 above)
Comment (LenkouAndrei) [reply movies-manager/src/containers/movie-form/movie-form.tsx]: Done

**Review by Vladlan — APPROVED** [review]

**Pull request merged**